### ISSUE TITLE
Check overwrite-policy before doing anything else.

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#5993]
Waiting for long build and then seeing is failing is frustrating

```
% cabal install .
Wrote tarball sdist to
/code/experiments/someprogram/dist-newstyle/sdist/someprogram-0.tar.gz
Resolving dependencies...
Build profile: -w ghc-8.4.4 -O1
In order, the following will be built (use -v for more details):
 - someprogram-0 (lib) (requires build)
 - someprogram-0 (exe:someprogram) (requires build)
Starting     someprogram-0 (lib)
Building     someprogram-0 (lib)
Installing   someprogram-0 (lib)
Completed    someprogram-0 (lib)
Starting     someprogram-0 (exe:someprogram)
Building     someprogram-0 (exe:someprogram)
Installing   someprogram-0 (exe:someprogram)
Completed    someprogram-0 (exe:someprogram)
Symlinking 'someprogram'
cabal: Path '/cabal/bin/someprogram' already exists. Use
--overwrite-policy=always to overwrite.
```

*EDIT* I consider this to be a bug.

**Comment by jneira** [comment]
This one should not be too difiicult to implement (i hope)

**Comment by fgaz** [comment]
Implemented in #9506
